src/cluster.c

```diff
@@ -5163,65 +5436,48 @@ void addNodeDetailsToShardReply(client *c, clusterNode *node) {
 }
 
 /* Add the shard reply of a single shard based off the given primary node. */
-void addShardReplyForClusterShards(client *c, clusterNode *node, uint16_t *slot_info_pairs, int slot_pairs_count) {
+void addShardReplyForClusterShards(client *c, list *nodes) {
+    serverAssert(listLength(nodes) > 0);
+    clusterNode *n = listNodeValue(listFirst(nodes));
     addReplyMapLen(c, 2);
     addReplyBulkCString(c, "slots");
-    if (slot_info_pairs) {
-        serverAssert((slot_pairs_count % 2) == 0);
-        addReplyArrayLen(c, slot_pairs_count);
-        for (int i = 0; i < slot_pairs_count; i++)
-            addReplyLongLong(c, (unsigned long)slot_info_pairs[i]);
+
+    /* Use slot_info_pairs from the primary only */
+    while (n->slaveof != NULL) n = n->slaveof;
+
+    if (n->slot_info_pairs != NULL) {
+        serverAssert((n->slot_info_pairs_count % 2) == 0);
+        addReplyArrayLen(c, n->slot_info_pairs_count);
+        for (int i = 0; i < n->slot_info_pairs_count; i++)
+            addReplyBulkLongLong(c, (unsigned long)n->slot_info_pairs[i]);
```

Comment:
i see we used to use `addReplyLongLong` and now it is `addReplyBulkLongLong`.
and i did not see we mentioned it in the top comment, is this expected?
if so, we should update the docs (in the doc its response is integer). https://redis.io/commands/cluster-shards/

Reply:
No, I think we should revert it back the previous behavior. It was probably unintended.

Reply:
#12561